lib/good_job/notifier.rb

```diff
@@ -112,5 +111,16 @@ def listen
     def listen_observer(_time, _result, _thread_error)
       listen unless shutdown?
     end
+
+    def with_listen_connection
+      ar_conn = ActiveRecord::Base.connection_pool.checkout.tap do |conn|
+        ActiveRecord::Base.connection_pool.remove(conn)
```

Comment:
Not a blocker, but I am curious: this will have the effect of using one _additional_ database connection than the pool size?

Reply:
Yes, it will, good spot. It should probably be documented. But I don't think this is unreasonable — the total number of ideal connections will be one for the notifier, and then one for each worker thread.

Reply:
Oh, I don't think they do — I can't find it. I'd expect it to be here:

https://edgeguides.rubyonrails.org/action_cable_overview.html#postgresql-adapter

Reply:
https://github.com/rails/rails/pull/40107